api/src/main/java/org/pmiops/workbench/api/OfflineUserController.java

```diff
@@ -74,9 +75,9 @@ private boolean timestampsEqual(Timestamp a, Timestamp b) {
 
         DbUser updatedUser;
         if (workbenchConfigProvider.get().featureFlags.enableMoodleV2Api) {
-          updatedUser = userService.syncComplianceTrainingStatusV2(user);
+          updatedUser = userService.syncComplianceTrainingStatusV2(user, AgentType.SYSTEM);
```

Comment:
Ideally we could keep the footprint smaller, so that callers don't need to know anything about the audit system. Would injecting the class @gjuggler mentioned help here? Could come later.

Reply:
Without a bigger refactoring of UserService, the best I can do here is the following:
```
syncComplianceTrainingStatusV2 ->
syncComplianceTrainingStatusAsCurrentUserV2()
syncComplianceTrainingStatusAsSystemV2()
```

I'm not sure this is much better.

Information about the actor needs to originate from the callsite, since UserService is currently overloaded with calls from various actors. I think architecturally we need to draw a line where request scoped information is attenuated in order to clean this up.

Reply:
OK. What would you. think about a small value class with the agent type, agent ID, and (optional) agent email builtin? That might save some lookups later on and ensure we don't accidentally pair a system agent with an email or invalid ID.

Reply:
Would you mind going ahead and filing a spike ticket for this question and the design sketch you mentioned about `RequestAuthenticationInfo`? I agree it's not a short term priority.

Reply:
Switched to passing a small value class. This eliminates the need for the Optional stuff entirely.

Spike is https://precisionmedicineinitiative.atlassian.net/browse/RW-4496